components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1980,6 +1980,42 @@ public void deleteIdP(String idPName, int tenantId, String tenantDomain)
         }
     }
 
+    public void forceDeleteIdP(String idPName,
+                               int tenantId,
+                               String tenantDomain) throws IdentityProviderManagementException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection();
+        try {
+            IdentityProvider identityProvider = getIdPByName(dbConnection, idPName, tenantId, tenantDomain);
+            if (identityProvider == null) {
```

Comment:
If the IDP name is incorrect is that fine to silently return without giving out an error to the caller ?

Reply:
We have done the same approach for deleteIdP() method as well so decided to do the same for forceDeleteIdp() method to make them consistent.

 If we decide to throw an exception that will be an API change IMO.